hudi-common/src/main/java/org/apache/hudi/common/config/TypedProperties.java

```diff
@@ -63,11 +64,22 @@ public Enumeration propertyNames() {
   public Set<String> stringPropertyNames() {
     Set<String> set = new LinkedHashSet<>();
     for (Object key : this.keys) {
-      set.add((String) key);
+      if (key instanceof String) {
+        set.add((String) key);
+      }
     }
     return set;
   }
 
+  public synchronized void putAll(Properties t) {
+    for (Map.Entry<?, ?> e : t.entrySet()) {
+      if (!containsKey(String.valueOf(e.getKey()))) {
+        keys.add(e.getKey());
+      }
+      super.put(e.getKey(), e.getValue());
```

Comment:
may be we can remove "super".

Reply:
What I'm thinking there about using `super.put` is that we have already add the keys into current keys(`LinkedHashset`), if we still call `this.put`, it will remove the key again and add the key later in the `LinkedHashset`, it seems that we don't need to update the key position in the `LinkedHashset`.

Reply:
Sorry, i get your point now. The key position in the set itself need not be updated.